nixos/modules/system/boot/loader/grub/grub.nix

```diff
@@ -137,6 +138,56 @@ in
         '';
       };
 
+      users = mkOption {
+        default = {};
+        example = {
+          root = { password_pbkdf2_file = "/path/to/file"; };
+        };
+        description = ''
+          User accounts for GRUB. When specified, the GRUB command line and
+          all boot options except the default are password-protected.
```

Comment:
It's probably good to add a warning that the passwords and hashes will be stored in the `/nix/store` world readable. And in `/boot/grub/grub.cfg`, I think.